### PR TITLE
Message Sender

### PR DESCRIPTION
**Pull request by Mikunj**
This is just the implementation PR. Test PR will be seperate as to not block others.

### CODE REVIEW

File: ts/session/sending/MessageSender.ts [hunk 1]
```diff
@@ -2,13 +2,110 @@
...
+): Promise<SignalService.Envelope> {
+  let source: string | undefined;
+  if (type !== SignalService.Envelope.Type.UNIDENTIFIED_SENDER) {
+    source = await UserUtil.getCurrentDevicePubKey();
+  }
+
+  return SignalService.Envelope.create({
+    type,
+    source,
+    sourceDevice: 1,
```
Comment (Bilb) [inline ts/session/sending/MessageSender.ts:53]: could we just get rid of this field in the transport layer? So we can progressively remove it from our code?

File: ts/session/sending/MessageSender.ts [hunk 2]
```diff
@@ -2,13 +2,110 @@
...
+  });
 }
 
+/**
+ * This is an outdated practice and we should probably just send the envelope data directly.
+ * Something to think about in the future.
+ */
+function wrapEnvelope(envelope: SignalService.Envelope): Uint8Array {
+  const request = SignalService.WebSocketRequestMessage.create({
+    id: 0,
```
Comment (Bilb) [inline ts/session/sending/MessageSender.ts:65]: existing code was this:
`    id: new Uint8Array(libsignal.crypto.getRandomBytes(1))[0], // random ID for now`

We do not care anymore?

File: ts/session/sending/MessageSender.ts [hunk 3]
```diff
@@ -2,13 +2,110 @@
...
+  const { group, quote, attachments, preview, body } = message;
+  const channelAPI = await lokiPublicChatAPI.findOrCreateChannel(
+    group.server,
+    group.channel,
+    group.conversationId
+  );
+
+  // Don't think returning true/false on `sendMessage` is a good way
+  // We should either: return nothing (success) or throw an error (failure)
+  return channelAPI.sendMessage({
```
Comment (Bilb) [inline ts/session/sending/MessageSender.ts:91]: I would prefer that way too (throwing errors)

File: ts/session/sending/MessageSender.ts (hunk 1 above)
Comment (Mikunj) [reply ts/session/sending/MessageSender.ts:53]: We can aim to do that but all platforms will need to be updated to handle the hardcoded value.

File: ts/session/sending/MessageSender.ts (hunk 2 above)
Comment (Mikunj) [reply ts/session/sending/MessageSender.ts:65]: Well we don't really use the id for much so I figured hardcoding it wouldn't matter.

**Pull request merged**